app/services/dataworks_mappers/redivis.rb

```diff
@@ -3,10 +3,110 @@
 module DataworksMappers
   # Map from Redivis metadata to Dataworks metadata
   class Redivis < Base
+    DATE_TYPES = [
+      { key: :temporalRange, type: 'Coverage' },
+      { key: :createdAt, type: 'Created' }
+    ].freeze
+
+    DESCRIPTION_TYPES = [
+      { key: :description, type: 'Abstract' },
+      { key: :methodologyMarkdown, type: 'Methods' },
+      { key: :usageMarkdown, type: 'Other' }
+    ].freeze
+
+    IDENTIFIER_TYPES = [
+      { key: :doi, type: 'DOI' },
+      { key: :qualifiedReference, type: 'RedivisReference' }
+    ].freeze
+
     def perform_map
       {
-        titles: [{ title: source[:name] }]
-      }
+        titles: [{ title: source[:name] }],
+        creators: [{ name: source[:owner][:fullName] }],
+        publication_year:,
+        identifiers:,
+        url: source[:url],
+        access:,
+        provider: 'Redivis'
+      }.merge(optional_params)
```

Comment:
Why merge in the optional params instead of just including here?

Reply:
Fixed. I moved this up into the perform_map... my thinking was that separating the optional for compacting from the required made sense, but if campacting all of them causes a validation error, that's probably what we'd want to see. The campacting is required to avoid insert empty arrays where a field is optional but if it's present has to have at least one.